datacube_stats/statistics.py

```diff
@@ -628,6 +628,136 @@ def __init__(self):
+        if wanted_names is None:
+            # default: include everything
+            return all_names
+
+        invalid = [name
+                   for name in wanted_names
+                   if name not in all_names]
+
+        if invalid:
+            msg = 'Specified measurements not found: {}'
+            raise StatsConfigurationError(msg.format(invalid))
+
+        return wanted_names
+
+    def measurements(self, input_measurements):
+        base = super(FCMedoid, self).measurements(input_measurements)
+
+        selected_names = self.select_names(self.output_measurements,
+                                           [m['name'] for m in base])
+
+        selected = [m for m in base if m['name'] in selected_names]
+
+        extra = [producer.measurement()
+                 for producer in self._metadata_producers]
+
+        return selected + extra
+
+    def medoid_indices(self, arr, invalid):
+        # vectorized version of `argnanmedoid`
+        bands, times, ys, xs = arr.shape
+
+        diff = (arr.reshape(bands, times, 1, ys, xs) -
+                arr.reshape(bands, 1, times, ys, xs))
+
+        dist = np.linalg.norm(diff, axis=0)
+        dist_sum = nansum(dist, axis=0)
+
+        dist_sum[invalid] = np.inf
+        return np.argmin(dist_sum, axis=0)
+
+    def compute(self, data):
+        # calculate medoid using only the fields in `input_measurements`
+        input_data = data[self.select_names(self.input_measurements,
+                                            list(data.data_vars))]
+
+        # calculate medoid indices
+        arr = input_data.to_array().values
+        invalid = anynan(arr, axis=0)
+        index = self.medoid_indices(arr, invalid)
+
+        # pixels for which there is not enough data
+        count_valid = np.count_nonzero(~invalid, axis=0)
+        not_enough = count_valid < self.minimum_valid_observations
+
+        # only report the measurements requested
+        output_data = data[self.select_names(self.output_measurements,
+                                             list(data.data_vars))]
+
+        def section_by_index(array, axis, index):
+            """
+            Take the slice of `var` indexed by entries of `index`
+            along the specified `axis`.
+            """
+            # alternative `axisindex` implementation
+
+            # possible index values for each dimension represented
+            # as numpy arrays all having the shape of `index`
+            indices = np.ix_(*[np.arange(dim) for dim in index.shape])
+
+            # the slice is taken along `axis`
+            # except for the array `index` itself, the other indices
+            # do nothing except trigger `numpy` fancy indexing
+            fancy_index = indices[:axis] + (index,) + indices[axis:]
+
+            # result has the same shape as `index`
+            return array[fancy_index]
+
+        def masked_section(var, axis, nodata):
```

Comment:
not too sure about the name, this seems to

1. Lookup pixel value according to previously computed Medoid index
2. Set output to `nodata` for "too few observations" case

If anything it's `extract_result` or something like that.